### PR TITLE
Continued header splitting

### PR DESCRIPTION
**Pull request by MCWertGaming**
Split the terminal.h into the first few headers. The rest requires a bit more work as the input needs the Terminal class and I have to fix that.

### CODE REVIEW

**Comment by certik** [comment]
Thanks. Once CI passes, please ping me and I'll review.

**Comment by MCWertGaming** [comment]
@certik the PR is done! I have done one file per commit to make it easier to review!

File: tests/test_terminal.cpp [hunk 1]
```diff
@@ -1,27 +1,27 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
 #include "doctest.h"
 
-#include <cpp-terminal/terminal.h>
+#include "../cpp-terminal/private/conversion.hpp"
```
Comment (certik) [inline tests/test_terminal.cpp:4]: This change shouldn't be necessary? I think the `<cpp-terminal/terminal.h>` should be the preferred way.

File: cpp-terminal/base.hpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <string>
+
+namespace Term {
+    enum class style {
```
Comment (certik) [inline cpp-terminal/base.hpp:6]: This should be indented as:
```suggestion

enum class style {
```
Also let's add an empty line.

File: cpp-terminal/prompt.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <cpp-terminal/window.hpp>
+#include <cpp-terminal/terminal.h>
+
+namespace Term {
+    // This model contains all the information about the state of the prompt in an
```
Comment (certik) [inline cpp-terminal/prompt.hpp:7]:
```suggestion
// This model contains all the information about the state of the prompt in an
```

File: cpp-terminal/window.hpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
+#pragma once
+
+#include <vector>
+#include <cpp-terminal/base.hpp>
+
+namespace Term {
+
+    /* Represents a rectangular window, as a 2D array of characters and their
```
Comment (certik) [inline cpp-terminal/window.hpp:8]:
```suggestion
/* Represents a rectangular window, as a 2D array of characters and their
```

**Review by certik — APPROVED** [review]
I think that generally looks very good. I left some minor suggestions.

File: tests/test_terminal.cpp (hunk 1 above)
Comment (MCWertGaming) [reply tests/test_terminal.cpp:4]: The problem is that the UTF-8 conversions are now in a seperate header file. The terminal.h header is removed soon.

**Comment by MCWertGaming** [comment]
@certik i'll merge this when we have resolved the pending discussion.

File: tests/test_terminal.cpp (hunk 1 above)
Comment (certik) [reply tests/test_terminal.cpp:4]: Why not `<cpp-terminal/private/terminal.h>`

**Comment by certik** [comment]
I think this is ok to merge now. The last item can be resolved in a new PR.

**Comment by MCWertGaming** [comment]
Moved the discussion into #131.

**Pull request merged**